04/SpeedTest.js

```diff
@@ -1,19 +1,60 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import useRandomItem from './hook';
 
-const SpeedTest = () => {
+function SpeedTest() {
     const [word, regenerateWord] = useRandomItem(['devmentor.pl', 'abc', 'JavaScript']);
+    const [counter, setCounter] = useState(0);
+    const [inputValue, setValue] = useState('');
+    const [result, setResult] = useState('');
+    const inputRef = useRef(null);
+    const finishRef = useRef(null);
 
+    const resetTimer = () => {
+        clearInterval(inputRef.current);
+        setCounter(counter - counter);
+    };
     useEffect(() => {
         regenerateWord();
     }, []);
+    useEffect(() => {
+        // eslint-disable-next-line no-console
+        console.log(word, inputValue);
+        if (word === inputValue) {
+            // eslint-disable-next-line no-alert
+            clearInterval(inputRef.current);
+            setResult(counter);
+            setCounter(0);
```

Comment:
Jeśli słowo jest poprawne to powinno być wylosowane nowe, czas ciągle się odlicza i możemy dalej pisać